components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java

```diff
@@ -473,31 +595,179 @@ public void testSetCookieNonExistCookieConfig() {
         verify(response, times(1)).addCookie(cookieCaptor.capture());
         List<Cookie> capturedCookies = cookieCaptor.getAllValues();
         Cookie storedCookie = capturedCookies.get(0);
-
         assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
         assertEquals(storedCookie.getPath(), ROOT_DOMAIN);
         assertEquals(storedCookie.getMaxAge(), age);
-
     }
 
     @Test
     public void testSetCookieExistCookieConfig() {
-        mockCookieTest();
 
+        mockCookieTest();
         IdentityCookieConfig cookieConfig = new IdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE);
-        mockStatic(IdentityUtil.class);
-        when(IdentityUtil.getIdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE)).thenReturn(cookieConfig);
+        IdentityUtil.getIdentityCookiesConfigurationHolder().put(FrameworkConstants.COMMONAUTH_COOKIE, cookieConfig);
         int age = 3600;
 
         FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age);
 
         verify(response, times(1)).addCookie(cookieCaptor.capture());
         List<Cookie> capturedCookies = cookieCaptor.getAllValues();
         Cookie storedCookie = capturedCookies.get(0);
-
         assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
         assertEquals(storedCookie.getPath(), ROOT_DOMAIN);
         assertEquals(storedCookie.getMaxAge(), age);
     }
 
+    @Test
+    public void testSetCookieWithSameSiteCookieNonExistCookieConfig() {
+
+        mockCookieTest();
+        int age = 3600;
+
+        FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age,
+                SameSiteCookie.STRICT, "Dummy-Path");
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie storedCookie = capturedCookies.get(0);
+        assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(storedCookie.getPath(), "Dummy-Path");
+        assertEquals(storedCookie.getMaxAge(), age);
+
```

Comment:
Unnecessary new line

Reply:
Addressed with commit c4e56d45a2072535df672c98b16637ada8cad476 in https://github.com/wso2/carbon-identity-framework/pull/3774 PR